submission/submission-webapp/src/main/kotlin/ac/uk/ebi/biostd/submission/web/resources/ext/ExtendedPageMapper.kt

```diff
@@ -1,29 +1,35 @@
 package ac.uk.ebi.biostd.submission.web.resources.ext
 
 import ac.uk.ebi.biostd.submission.web.model.ExtPage
+import ac.uk.ebi.biostd.submission.web.model.ExtPageRequest
 import ebi.ac.uk.extended.model.ExtSubmission
 import org.springframework.data.domain.Page
 import org.springframework.data.domain.Pageable
+import org.springframework.web.util.UriComponentsBuilder
 import java.net.URI
 
 class ExtendedPageMapper(private val instanceBase: URI) {
-
-    fun asExtPage(page: Page<ExtSubmission>): ExtPage =
+    fun asExtPage(page: Page<ExtSubmission>, request: ExtPageRequest): ExtPage =
         ExtPage(
             content = page.content,
             totalElements = page.totalElements,
             offset = page.pageable.offset,
             limit = page.pageable.pageSize,
-            next = getNext(page),
-            previous = getPrevious(page)
+            next = getNext(page, request),
+            previous = getPrevious(page, request)
         )
 
-    private fun getPrevious(page: Page<ExtSubmission>): String? =
-        if (page.hasPrevious()) instanceBase.resolve(asQueryParams(page.previousPageable())).toString() else null
+    private fun getPrevious(page: Page<ExtSubmission>, request: ExtPageRequest): String? =
+        if (page.hasPrevious()) instanceBase.resolve(asUrl(page.previousPageable(), request)).toString() else null
 
-    private fun getNext(page: Page<ExtSubmission>): String? =
-        if (page.hasNext()) instanceBase.resolve(asQueryParams(page.nextPageable())).toString() else null
+    private fun getNext(page: Page<ExtSubmission>, request: ExtPageRequest): String? =
+        if (page.hasNext()) instanceBase.resolve(asUrl(page.nextPageable(), request)).toString() else null
 
-    private fun asQueryParams(nextPageable: Pageable): String =
-        "$instanceBase/submissions/extended?offset=${nextPageable.offset}&limit=${nextPageable.pageSize}"
+    private fun asUrl(next: Pageable, request: ExtPageRequest): String =
+        UriComponentsBuilder.fromUriString("$instanceBase/submissions/extended").apply {
+            queryParam("offset", next.offset)
+            queryParam("limit", next.pageSize)
+            request.fromRTime?.let { queryParam("fromRTime", it) }
+            request.toRTime?.let { queryParam("toRTime", it) }
+        }.build().toString()
```

Comment:
```
    private fun asUrl(next: Pageable, request: ExtPageRequest): String =
        UriComponentsBuilder.fromUriString("$instanceBase/submissions/extended")
            .queryParam("offset", next.offset)
            .queryParam("limit", next.pageSize)
            .optionalQueryParam("fromRTime", request.fromRTime)
            .optionalQueryParam("toRTime", request.toRTime)
            .build()
            .toUriString()
```


this looks like a strange implementation because it a builder and we are using apply. 
```
    private fun UriComponentsBuilder.optionalQueryParam(name: String, value: Any?): UriComponentsBuilder {
        value?.let { queryParam(name, it) }
        return this
    }
```

Reply:
I agree that it looks better on an extension but I don't think using `apply` makes it confusing since it's intended to do something with an object and we're basically doing the same in the extension function 😅. I'll make the change anyway.

Reply:
yes I agree is not more confusing but this is a builder class so what one will expect is to be used as a builder. Extension favor that.

Reply:
Done